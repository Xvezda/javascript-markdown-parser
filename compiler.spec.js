const tokenizer = require('./tokenizer.js');
const lexer = require('./lexer.js');
const parser = require('./parser.js');
const compiler = require('./compiler.js');

function mdToHtml(md) {
  const tokens = lexer(tokenizer(md));
  const ast = parser(tokens);
  const html = compiler(ast);
  return html;
}

test('복합 마크다운 변환', () => {
  const md = `\
# hello world
## hi

foobar
---
egg
ham
spam

link to [Google](http://google.com)`;

  expect(mdToHtml(md)).toMatchSnapshot();
});

test('제목 변환', () => {
  expect(mdToHtml('# foo')).toMatch('<h1>foo</h1>');
  expect(mdToHtml('## foo')).toMatch('<h2>foo</h2>');
  expect(mdToHtml('### foo')).toMatch('<h3>foo</h3>');
  expect(mdToHtml('#### foo')).toMatch('<h4>foo</h4>');
  expect(mdToHtml('##### foo')).toMatch('<h5>foo</h5>');
  expect(mdToHtml('###### foo')).toMatch('<h6>foo</h6>');

  const h1 = `\
foo
===
`;
  expect(mdToHtml(h1)).toMatch('<h1>foo</h1>')

  const h2 = `\
foo
---
`;
  expect(mdToHtml(h2)).toMatch('<h2>foo</h2>');
});

test('올바르지 않은 제목 문법 처리', () => {
  const noSpace = '#foo';
  expect(mdToHtml(noSpace)).toMatchSnapshot();

  const invalidLine = `\
foo
---=
`
  expect(mdToHtml(invalidLine)).toMatchSnapshot();
});

test('줄바꿈', () => {
  const br = `\
foo
bar
`;
  expect(mdToHtml(br)).toMatchSnapshot();

  const p = `\
foo

bar
`;
  expect(mdToHtml(p)).toMatchSnapshot();

  const nop = `\



foo
`;
  expect(mdToHtml(nop)).toMatchSnapshot();

});

test('링크', () => {
  expect(mdToHtml('[google](https://www.google.com)')).toMatchSnapshot();
});

test('이미지', () => {
  expect(mdToHtml('![random image](https://unsplash.it/100)'))
    .toMatchSnapshot();
});
