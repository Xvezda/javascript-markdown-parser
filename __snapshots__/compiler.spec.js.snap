// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`링크 1`] = `<p><a href="https://www.google.com">google</a></p>`;

exports[`복합 마크다운 변환 1`] = `
<h1>hello world</h1>
<h2>hi</h2>
<h2>foobar</h2>
<p>egg<br />ham<br />spam</p>
<p>link to <a href="http://google.com">Google</a></p>
`;

exports[`올바르지 않은 제목 문법 처리 1`] = `<p>#foo</p>`;

exports[`올바르지 않은 제목 문법 처리 2`] = `<p>foo<br />---=</p>`;

exports[`이미지 1`] = `<p><img src="https://unsplash.it/100" alt="random image" /></p>`;

exports[`줄바꿈 1`] = `<p>foo<br />bar</p>`;

exports[`줄바꿈 2`] = `
<p>foo</p>
<p>bar</p>
`;

exports[`줄바꿈 3`] = `<p>foo</p>`;
